ios/browser/api/sync/brave_sync_api.mm

```diff
@@ -0,0 +1,203 @@
+#include "ios/chrome/browser/sync/sync_setup_service.h"
+#include "ios/chrome/browser/sync/sync_setup_service_factory.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+
+
+@interface BraveSyncDeviceObserver()
+{
+  std::unique_ptr<BraveSyncDeviceTracker> _device_observer;
+}
+@end
+
+@implementation BraveSyncDeviceObserver
+
+- (instancetype)initWithCallback:(void(^)())onDeviceInfoChanged {
+  if ((self = [super init])) {
+    _device_observer = std::make_unique<BraveSyncDeviceTracker>(onDeviceInfoChanged);
+  }
+  return self;
+}
+@end
+
+@interface BraveSyncServiceObserver()
+{
+  std::unique_ptr<BraveSyncServiceTracker> _service_observer;
+}
+@end
+
+@implementation BraveSyncServiceObserver
+
+- (instancetype)initWithCallback:(void(^)())onSyncServiceStateChanged {
+  if ((self = [super init])) {
+    _service_observer = std::make_unique<BraveSyncServiceTracker>([onSyncServiceStateChanged](syncer::SyncService *sync) {
+      onSyncServiceStateChanged();
+    }, [](syncer::SyncService *sync) {
+        fprintf(stderr, "Sync Shut Down\n");
+    });
+  }
+  return self;
+}
+@end
+
+@interface BraveSyncAPI()
+{
+  std::unique_ptr<BraveSyncWorker> _worker;
+}
+@end
+
+@implementation BraveSyncAPI
+
++ (instancetype)sharedSyncAPI {
+	static BraveSyncAPI *instance = nil;
+	static dispatch_once_t onceToken;
+	dispatch_once(&onceToken, ^{
+		instance = [[BraveSyncAPI alloc] init];
+	});
+	return instance;
+}
+
+- (instancetype)init {
+  if ((self = [super init])) {
+    ios::ChromeBrowserStateManager* browserStateManager =
+        GetApplicationContext()->GetChromeBrowserStateManager();
+    ChromeBrowserState* chromeBrowserState =
+        browserStateManager->GetLastUsedBrowserState();
+    _worker.reset(new BraveSyncWorker(chromeBrowserState));
+  }
+  return self;
+}
+
+- (void)dealloc {
+  _worker.reset();
+}
+
+- (bool)syncEnabled {
+  return _worker->IsSyncEnabled();
+}
+
+- (void)setSyncEnabled:(bool)enabled {
+  _worker->SetSyncEnabled(enabled);
+}
+
+- (bool)isSyncFeatureActive {
+  return _worker->IsSyncFeatureActive();
+}
+
+- (bool)isValidSyncCode:(NSString *)syncCode {
+  return _worker->IsValidSyncCode(base::SysNSStringToUTF8(syncCode));
+}
+
+- (NSString *)getSyncCode {
+  std::string syncCode = _worker->GetOrCreateSyncCode();
+  if (syncCode.empty()) {
+    return nil;
+  }
+
+  return base::SysUTF8ToNSString(syncCode);
+}
+
+- (bool)setSyncCode:(NSString *)syncCode {
+  return _worker->SetSyncCode(base::SysNSStringToUTF8(syncCode));
+}
+
+- (NSString *)syncCodeFromHexSeed:(NSString *)hexSeed {
+  return base::SysUTF8ToNSString(_worker->GetSyncCodeFromHexSeed(base::SysNSStringToUTF8(hexSeed)));
+}
+
+- (UIImage *)getQRCodeImage:(CGSize)size {
+  std::vector<uint8_t> seed;
+  std::string sync_code = _worker->GetOrCreateSyncCode();
+  if (!brave_sync::crypto::PassphraseToBytes32(sync_code, &seed)) {
+    return nil;
+  }
+
```

Comment:
probably should throw an error here if `seed` is not the expected size (32 bytes)

Reply:
On the iOS side, we check if the seed is the expected side first because calling this function. However, this code is taken from the current merged sync code in master and is used on Android as well.